apps/meteor/client/providers/AppsProvider.tsx

```diff
@@ -133,9 +133,18 @@ const AppsProvider: FC = ({ children }) => {
 		<AppsContext.Provider
 			children={children}
 			value={{
-				installedApps: { phase: AsyncStatePhase.RESOLVED, value: { apps: store.data?.[1] || [] } },
-				marketplaceApps: { phase: AsyncStatePhase.RESOLVED, value: { apps: store.data?.[0] || [] } },
-				privateApps: { phase: AsyncStatePhase.RESOLVED, value: { apps: store.data?.[2] || [] } },
+				installedApps: {
+					phase: store.isLoading ? AsyncStatePhase.LOADING : AsyncStatePhase.RESOLVED,
+					value: { apps: store.data?.[1] || [] },
+				},
+				marketplaceApps: {
+					phase: store.isLoading ? AsyncStatePhase.LOADING : AsyncStatePhase.RESOLVED,
+					value: { apps: store.data?.[0] || [] },
+				},
+				privateApps: {
+					phase: store.isLoading ? AsyncStatePhase.LOADING : AsyncStatePhase.RESOLVED,
+					value: { apps: store.data?.[2] || [] },
+				},
```

Comment:
How about extracting the common logic into a function, and destructing the array with more descriptive variables, something like, below:

```ts
const getAppState = (apps) => ({
  phase: store.isLoading ? AsyncStatePhase.LOADING : AsyncStatePhase.RESOLVED,
  value: { apps: apps || [] },
});

const [marketplaceAppsData, installedAppsData, privateAppsData] = store.data || [];

<AppsContext.Provider
  children={children}
  value={{
    installedApps: getAppState(installedAppsData),
    marketplaceApps: getAppState(marketplaceAppsData),
    privateApps: getAppState(privateAppsData),
    reload: async () => {
      await Promise.all([queryClient.invalidateQueries(['marketplace'])]);
    },
  }}
/>
```

Reply:
Thanks for the suggestion, nice catch! <3